### PR TITLE
Added google authorization

### PR DESCRIPTION
**Pull request by himanshuc3**
Adds Google Authorization and solves #49 .

### CODE REVIEW

**Comment by vigneshtdev** [comment]
@himanshuc3 dude how did you get past the newrelic.js error?

File: src/passport/strategies/user/index.js [hunk 1]
```diff
@@ -12,4 +12,6 @@ const githubStrategy = require('./strategy-github')
 
 const lmsStrategy = require('./strategy-lms')
 
-module.exports = {localStrategy, fbStrategy, twitterStrategy, githubStrategy, lmsStrategy}
\ No newline at end of file
+const googleStrategy = require('./strategy-google')
```
Comment (championswimmer) [inline src/passport/strategies/user/index.js]: do it the same way as other strategies are done

File: src/server.js [hunk 2]
```diff
@@ -1,7 +1,7 @@
 /**
  * Created by championswimmer on 08/03/17.
  */
-require('newrelic')
+// require('newrelic')
```
Comment (championswimmer) [inline src/server.js]: can't stop newrelic in production. you might have neeeded it for dev. please uncomment

File: .gitignore [hunk 3]
```diff
@@ -47,6 +47,7 @@ typings/
 .env
 secrets.json
 .github-todos
+keys.js
```
Comment (championswimmer) [inline .gitignore]: why? what is in keys.js

Comment (himanshuc3) [reply .gitignore]: my cliend id and secret

File: src/server.js (hunk 2 above)
Comment (himanshuc3) [reply src/server.js]: done.

File: .gitignore (hunk 3 above)
Comment (championswimmer) [reply .gitignore]: remove that from .gitignore. I don't care about your keys :D

File: src/passport/strategies/user/strategy-google.js [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+const passutils = require('../../../utils/password') // hashing support by bcrypt
+
+/**
+ * Authenticate _users_ using their Twitter accounts
+ */
+
+module.exports = new GoogleStrategy({
+    clientID: keys.googleClientID,
+    clientSecret: keys.googleClientSecret,
+    callbackURL: config.SERVER_URL + config.GITHUB_CALLBACK,
```
Comment (championswimmer) [inline src/passport/strategies/user/strategy-google.js]: it should'nt be GITHUB_CALLBACK here

File: src/server.js (hunk 2 above)
Comment (championswimmer) [reply src/server.js]: I don't see it being uncommented ?

File: src/passport/strategies/user/strategy-google.js [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+ * Authenticate _users_ using their Twitter accounts
+ */
+
+module.exports = new GoogleStrategy({
+    clientID: secrets.GOOGLE_CLIENT_ID,
+    clientSecret: secrets.GOOGLE_CLIENT_SECRET,
+    callbackURL: config.SERVER_URL + config.GOOGLE_CALLBACK,
+    passReqToCallback: true
+}, (req, accessToken, refreshToken, profile, cb) =>{
+    // console.log('Accestoken:',accessToken)
```
Comment (championswimmer) [inline src/passport/strategies/user/strategy-google.js]: remove commented lines

File: src/passport/strategies/user/strategy-google.js [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+                        user: {
+                            username: existCount === 0 ? profileJson.login : profileJson.login + "-gh",
+                            firstname: profileJson.name ? profileJson.name.split(' ')[0] : profileJson.login,
+                            lastname: profileJson.name ? profileJson.name.split(' ').pop() : profileJson.login,
+                            email: profileJson.email,
+                            photo: profileJson.avatar_url
+                        }
+                    }
+                })
+            }).spread(function (userGithub, created) {
```
Comment (championswimmer) [inline src/passport/strategies/user/strategy-google.js]: userGoogle, not userGithub

File: src/passport/strategies/user/strategy-google.js [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+            //TODO: Check created == true for first time
+            if (!userGithub) {
```
Comment (championswimmer) [inline src/passport/strategies/user/strategy-google.js]: userGoogle not userGithub

File: src/passport/strategies/user/strategy-google.js [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+                return cb(null, false, {message: 'Authentication Failed'})
+            }
+            return cb(null, userGoogle.user.get())
```
Comment (championswimmer) [inline src/passport/strategies/user/strategy-google.js]: did you even test your code? you use userGoogle at some places and userGithub at others. obviously this entire if-else branch does not work as intended at all

Comment (himanshuc3) [reply src/passport/strategies/user/strategy-google.js]: I think i did something wrong with the git commands. These small typos, I already fixed before. I would get back to you after all these silly mistakes.

**Pull request closed without merging**